shotgun_api3/shotgun.py

```diff
@@ -3853,6 +3851,23 @@ def _complete_multipart_upload(self, upload_info, filename, etags):
         if not str(result).startswith("1"):
             raise ShotgunError("Unable get upload part link: %s" % result)
 
+    def _requires_direct_s3_upload(self, entity_type, field_name):
+        """
+        Internal function that determines if an entity_type + field_name combination
+        should be uploaded to cloud storage.
+
+        :param str entity_type: The entity type of the file being uploaded.
+        :param str field_name: The matching field name for the file being uploaded.
+
+        :returns: Whether the field + entity type combination should be uploaded to cloud storage.
+        :rtype: bool
+        """
+        supported_s3_types = self.server_info.get('s3_enabled_upload_types', {})
```

Comment:
@daigles - just double checking - this code currently doesn't do any shotgun server version checking - are we safe to assume that all versions of shotgun is returning the same data structure for `s3_enabled_upload_types`?

Reply:
@daigles - are we ok to use `s3` here in our code and flags? IIRC, the code that you originally put in was trying to avoid mentioning what backend storage solution was being used (since this is an implementation detail)! Now it looks like we are starting to put it in. Would be nice if we could continue to keep the code cloud-agnostic.

Reply:
Good point. This code needs to work with previous versions of SG backend... The default value should correspond with what we currently support (e.g. Version, sg_uploaded_movie).